saleor/product/utils.py

```diff
@@ -97,32 +99,61 @@ def products_for_cart(user):
     return products
 
 
-def get_variant_picker_data(variants, attributes):
-    data = {'attributes': [], 'variants': []}
-    for attribute in attributes:
-        data['attributes'].append({
+def get_variant_picker_data(product, discounts=None):
+    availability = get_availability(product, discounts)
+    variants = product.variants.all()
+    data = {'productAttributes': [], 'variantAttributes': [], 'variants': []}
+
+    variant_attributes = product.product_class.variant_attributes.all()
+    for attribute in variant_attributes:
+        data['variantAttributes'].append({
             'pk': attribute.pk,
             'display': attribute.display,
             'name': attribute.name,
             'values': [{'pk': value.pk, 'display': value.display}
-                       for value in attribute.values.all()]
-        })
+                       for value in attribute.values.all()]})
+
+    product_attributes = get_product_attributes_data(product)
+    for attribute, value in product_attributes.items():
+        data['productAttributes'].append({
+            'attribute': attribute.display,
+            'value': value.display})
+
     for variant in variants:
-        price = variant.get_price_per_item()
+        price = variant.get_price_per_item(discounts)
+        price_undiscounted = variant.get_price_per_item()
         variant_data = {
             'id': variant.id,
             'price': float(price.gross),
+            'priceUndiscounted': float(price_undiscounted.gross),
```

Comment:
I'm a bit worried about the floats here. Would prefer a JSON serializer that knew how to handle Decimals.